pkg/istio/eventingistio/istio.go

```diff
@@ -132,3 +142,42 @@ func serviceMeshNetworkPolicies() []networkingv1.NetworkPolicy {
 		},
 	}
 }
+
+func MaybeScaleIstioController(client client.Client, eventing *operatorv1beta1.KnativeEventing) error {
+	if enabled := IsEnabled(eventing.GetSpec().GetConfig()); enabled {
+		return scaleEventingIstioController(client, eventing, func(d *appsv1.Deployment) (bool, int32) {
+			return *d.Spec.Replicas < 1, 1
+		})
+	}
+	return scaleEventingIstioController(client, eventing, func(d *appsv1.Deployment) (bool, int32) {
+		return *d.Spec.Replicas > 0, 0
+	})
+}
+
+func scaleEventingIstioController(client client.Client, eventing *operatorv1beta1.KnativeEventing, shouldScaleFn func(d *appsv1.Deployment) (bool, int32)) error {
+	istioControllerName := types.NamespacedName{Namespace: eventing.GetNamespace(), Name: "eventing-istio-controller"}
+
+	overrides := eventing.GetSpec().GetWorkloadOverrides()
```

Comment:
Just curious, we already have product documentation for the workload ovrrides?

But, good using that, instread of ha-replica